modules/nw-sriov-example-dpdk-line-rate.adoc

```diff
@@ -0,0 +1,279 @@
+
+There are some differences between running DPDK workloads on Intel VFs or Mellanox VFs. We will look at both cases.
+
+The following is an example of an `sriovNetworkNodePolicy` used to run DPDK applications on Intel NICs:
+[source,yaml]
+----
+    apiVersion: sriovnetwork.openshift.io/v1
+    kind: SriovNetworkNodePolicy
+    metadata:
+     name: dpdk-nic-1
+     namespace: openshift-sriov-network-operator
+    spec:
+     deviceType: vfio-pci
+     needVhostNet: true
+     nicSelector:
+       pfNames: ["ens3f0"]
+     nodeSelector:
+       node-role.kubernetes.io/worker-cnf: ""
+     numVfs: 10
+     priority: 99
+     resourceName: dpdk_nic_1
+    ---
+    apiVersion: sriovnetwork.openshift.io/v1
+    kind: SriovNetworkNodePolicy
+    metadata:
+     name: dpdk-nic-1
+     namespace: openshift-sriov-network-operator
+    spec:
+     deviceType: vfio-pci <1>
+     needVhostNet: true <2>
+     nicSelector:
+         pfNames: ["ens3f1"]
+     nodeSelector:
+     node-role.kubernetes.io/worker-cnf: ""
+     numVfs: 10
+     priority: 99
+     resourceName: dpdk_nic_2
+
+----
+<1> For Intel NICs, `deviceType` must be `vfio-pci`.
+<2> If kernel communication with DPDK workloads is required, add `needVhostNet: true`. This will mount the `/dev/net/tun` and `/dev/vhost-net` devices into the container so the application can create a tap device and connect the tap device to the DPDK workload.
+
+== Using `sriovNetworkNodePolicy` to run DPDK applications on Mellanox NICs
+The following is an example of `sriovNetworkNodePolicy` for Mellanox NICs:
+[source,yaml]
+----
+    apiVersion: sriovnetwork.openshift.io/v1
+    kind: SriovNetworkNodePolicy
+    metadata:
+     name: dpdk-nic-1
+     namespace: openshift-sriov-network-operator
+    spec:
+     deviceType: netdevice <1>
+     isRdma: true
+     nicSelector:
+       rootDevices:
+         - "0000:5e:00.1"
+     nodeSelector:
+       node-role.kubernetes.io/worker-cnf: ""
+     numVfs: 5
+     priority: 99
+     resourceName: dpdk_nic_1
+    ---
+    apiVersion: sriovnetwork.openshift.io/v1
+    kind: SriovNetworkNodePolicy
+    metadata:
+     name: dpdk-nic-2
+     namespace: openshift-sriov-network-operator
+    spec:
+     deviceType: netdevice <1>
+     isRdma: true <2>
+     nicSelector:
+       rootDevices:
+         - "0000:5e:00.0"
+     nodeSelector:
+       node-role.kubernetes.io/worker-cnf: ""
+     numVfs: 5
+     priority: 99
+     resourceName: dpdk_nic_2
+----
+<1> For Mellanox devices the `deviceType` must be `netdevice`.
+<2> For Mellanox devices `isRDMA` must be `true`.
+Mellanox cards are connected to DPDK applications using Flow Bifurcation. This mechanism splits traffic between Linux user space and kernel space, and can enhance line rate processing capability.
+
+== Using the SR-IOV network operator to create the `sriovNetwork`
+The following is an example definition of an `sriovNetwork`. In this case Intel and Mellanox configurations are identical.
```

Comment:
Should this be "an `sriovNetwork` _object_" or maybe, as below, "an `sriovNetwork` _CR_"?
If so would also apply to line above.